### DISCUSSION TITLE
Queries about the "Baseline" feature

### DISCUSSION THREAD

**Discussion by OnkarRuikar** [mdn/mdn-community#392] [Platform] [answered]
We are showing [baseline banner](https://developer.mozilla.org/en-US/docs/Web/CSS/grid) on some pages. I've a few queries regarding the feature on MDN website.
1. When I see the baseline banner on [`flex-flow` page](https://web.archive.org/web/20230521000041mp_/https://developer.mozilla.org/en-US/docs/Web/CSS/flex-flow), does it mean the property is baseline or the entire `flexbox` module baseline? Asking this question because [the data source has assigned](https://github.com/web-platform-dx/feature-set/blob/main/feature-group-definitions/flexbox.yml) the baseline status to the `flexbox` module not the property.
2. When I don't see the banner on a flexbox property page, say [`justify-content`](https://web.archive.org/web/20230520234906/https://developer.mozilla.org/en-US/docs/Web/CSS/justify-content) does it mean the feature/property is not baseline?
3. How absence of baseline header is different than [yellow baseline status](https://web.archive.org/web/20230521001245/https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Subgrid)?
4. Our BCD queries don't match with the Baseline data source (feature-set). For example. ` css.properties.justify-content (BCD)` vs `css.properties.justify-content.flex_context (feature-set)`. Maybe that is why `justify-content` page is not showing the banner. How are we going to map their queries with ours?
5. They haven't classified the feature documents CSS, HTML, JS etc. wise. They've dumped all of them in the same root [`webFeatures` array](https://github.com/web-platform-dx/feature-set/tree/main/feature-group-definitions). When they complete adding thousands of web features/apis/props there, our current implementation will be in trouble:
```js
// for each doc
function addBaseline(doc: Partial<Doc>): WebFeatureStatus | undefined {
  if (doc.browserCompat) {
    for (const feature of Object.values<WebFeature>(webFeatures)) {
      if (
        feature.status &&
        feature.compat_features?.some((query) =>
          doc.browserCompat?.includes(query)
        )
      ) {
        return feature.status;
      }
    }
  }
}
```
 We need to ask them for classification and reverse lookup.
7. If the definition of baseline is
> The Web Platform Baseline consists of features natively supported in the core browser set for at least two major versions. 
>     -- [the blog](https://developer.mozilla.org/en-US/blog/baseline-unified-view-stable-web-features/#the_web_platform_baseline)

... then why don't we infer the status from our already available BCD data? If baselines are being built using BCD itself, then why don't we use BCD directly? I'll solve issues in point 4 and 5.
8. If both baseline and BCD are statuses then can we combine them in one section on a MDN page? It doesn't make sense sending reader from one end of the page to the other using `See full compatibility` hyperlink.

**Comment by tianheg** [comment]
After this blog ([`Introducing Baseline: a unified view of stable web features`](https://developer.mozilla.org/en-US/blog/baseline-unified-view-stable-web-features/)) came out, someone built this extension [romainmenke/mdn-support-context](https://github.com/romainmenke/mdn-support-context). I'd like to get an official implementation about it.

**Reply by OnkarRuikar** [reply]
Its personalized baseline using browserlist query is a cool feature :rocket:, as not everyone targets same browsers.\
But they've packed entire 3MB BCD json in the extension, which will get stale pretty quickly. And there is no option to update it.

**Comment by schalkneethling** [comment]
Thank you for starting the conversation, @OnkarRuikar. Tagging, @Rumyra and @atopal

**Comment by Rumyra** [comment, marked as answer]
Hi @OnkarRuikar thanks for opening the discussion - here are hopefully some insights:

For all of the queries it would be good to point out this is a prototype and work in progress, over the next few months we plan to improve the feature and roll it out across the site.

1. Its for the flexbox module. One thing to be clear about with baseline is that it is based on a set of data called 'feature-sets' which is a project run by the Web Developer Experience Community Group to create a common language for web features. Support is just a part of what that can be used for (and in this first instance for 'baseline') - I do think making it more clear that this is for the 'set' rather than the property is great feedback, thank you
2. As we haven't rolled it out site wide yet it means we haven't implemented it on a page - we'll be working on this in the near future
3. As above - we're in the process of rolling it out but also iterating on the feature as well, based on feedback. If there is a yellow icon, at the moment that means not widely supported (subject to change)
4. We're currently considering this. BCD is included in the feature sets, but as we know bcd doesn't map directly to all pages which coul dhave baseline, we're exploring options
5. This is because a set of features to describe could include multiple technologies
7. As mentioned this is about groups of features and could be used for other functionality as well as support. We are however using bcd as the support status (it's intention)
8. Thank you for this feedback also - the original intent was to give readers a quick immediate view, rather than the more complex view of the bcd table

I hope that helps somewhat!

**Reply by OnkarRuikar** [reply]
>  this is for the 'set' rather than the property

So, a feature-set includes only the features(e.g. properties) that are "baseline" themselves. And if specs add a new property to the feature then it won't be in the set till it is well supported as per the definition, right?\
There won't be a banner for that property page or the banner will be yellow.

**Reply by Rumyra** [reply]
It's sort of the other way around - a feature-set is independant of support, it's just a group of things that combine to make a 'feature' - like CSS Grid would include all the properties. If things included in the feature set are at a certain level of support, then that feature is described as baseline - I'm sure you've seen it, but the readme on the repo explains it well https://github.com/web-platform-dx/feature-set 👍
